configure.ac

```diff
@@ -111,13 +119,31 @@ if test "x$asan_enabled" = "xtrue"; then
     AC_SUBST(LDFLAGS_ASAN)
 fi
 
+if test "x$python2" = "xtrue"; then
+	m4_define_default([_AM_PYTHON_INTERPRETER_LIST],
+					  [python2 python2.7 python2.6 python2.5 python2.4 python2.3 python2.2 python2.1 python2.0])
+	AM_PATH_PYTHON
+	AX_PYTHON_DEVEL([< '3'])
+	PYTHON2_CPPFLAGS=$PYTHON_CPPFLAGS
+	PYTHON2_LIBS=$PYTHON_LIBS
+	AC_SUBST(PYTHON2_CPPFLAGS)
+	AC_SUBST(PYTHON2_LIBS)
+else
+	# The py-compile script either uses $PYTHON (which won't be set if we're not building python 2 bindings),
+	# or assumes a default of `python` (which may or may not exist as a symlink). Therefore, just assign
+	# PYTHON3 to PYTHON to make sure it always points to a valid binary.
+	PYTHON=$PYTHON3
+fi
+
 AM_CONDITIONAL(ASAN_ENABLED, test x$asan_enabled = xtrue)
+AM_CONDITIONAL(PYTHON2, test x$python2 = xtrue)
 
 AC_PATH_PROGS(SWIG, [swig3.0 swig])
 
 CXXFLAGS_COMMON=""
 CXXFLAGS_COMMON+=" -ansi"
 CXXFLAGS_COMMON+=" -fPIC"
+CXXFLAGS_COMMON+=" -pipe"
```

Comment:
why this is added? something wrong with temporary files ?

Reply:
Hmm, I don't remember why I added this. At best, this might give a bit of performance/space benefit since intermediate files will all get transferred in memory rather than needing to be saved to a file first. Using temporary files shouldn't cause any issue, unless disk space on `/tmp` is very low.